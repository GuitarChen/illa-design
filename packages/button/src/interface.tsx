import { ButtonHTMLAttributes, HTMLAttributes, ReactNode } from "react"
import { SerializedStyles } from "@emotion/react"

export type ButtonColorScheme =
  | "white"
  | "blackAlpha"
  | "gray"
  | "grayBlue"
  | "red"
  | "orange"
  | "yellow"
  | "green"
  | "blue"
  | "cyan"
  | "purple"
  | "techPink"
  | "techPurple"

export type ButtonSize = "small" | "medium" | "large"
export type ButtonVariant = "fill" | "dashed" | "outline" | "text" | "light"
export type ButtonShape = "square" | "round"

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  _css?: SerializedStyles
  colorScheme?: ButtonColorScheme
  size?: ButtonSize
  variant?: ButtonVariant
  shape?: ButtonShape
  fullWidth?: boolean
  loading?: boolean
  loadingText?: string
  leftIcon?: ReactNode
  rightIcon?: ReactNode
  buttonRadius?: string
  autoFullVertically?: boolean
  autoFullHorizontal?: boolean
  borderColor?: string
  backgroundColor?: string
  textColor?: string
}

export interface ButtonGroupProps
  extends HTMLAttributes<HTMLDivElement>,
    Omit<ButtonGroupContextProps, "first" | "last"> {}

export interface ButtonGroupContextProps {
  colorScheme?: ButtonColorScheme
  size?: ButtonSize
  variant?: ButtonVariant
  shape?: ButtonShape
  spacing?: string
  attached?: boolean
  first?: boolean
  last?: boolean
}
