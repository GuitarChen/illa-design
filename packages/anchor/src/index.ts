export * from "./anchor"
export * from "./interface"
