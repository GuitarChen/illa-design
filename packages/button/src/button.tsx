import { Children, forwardRef, useMemo } from "react"
import { ButtonProps } from "./interface"
import { css } from "@emotion/react"
import { LoadingIcon } from "@illa-design/icon"
import {
  applyBg,
  applyCursor,
  applyElementColor,
  applyFontStyle,
  applyIconWithoutText,
  applyLeftIconStyle,
  applyPaddingStyle,
  applyRightIconStyle,
  applyShape,
  applyTagContainer,
  applyWithoutTextSize,
} from "./style"
import { ButtonGroupContext } from "."
import { getSizeCssByAutoFullProps } from "@illa-design/system"

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => {
    const { autoFullHorizontal, autoFullVertically } = props
    const sizeCss = useMemo(
      () => getSizeCssByAutoFullProps(autoFullHorizontal, autoFullVertically),
      [autoFullHorizontal, autoFullVertically],
    )

    return (
      <ButtonGroupContext.Consumer>
        {(value) => {
          const { attached, first, last } = value ?? {}

          const {
            _css,
            colorScheme = value?.colorScheme ?? "blue",
            size = value?.size ?? "medium",
            variant = value?.variant ?? "fill",
            shape = value?.shape ?? "square",
            fullWidth,
            loading,
            loadingText,
            leftIcon,
            disabled,
            rightIcon,
            buttonRadius,
            borderColor,
            backgroundColor,
            textColor,
            autoFullHorizontal,
            autoFullVertically,
            ...otherProps
          } = props

          const hasLoadingText =
            loadingText != undefined && loadingText.length > 0
          const hasChild = Children.count(props.children) >= 1
          const hasPropContent = hasChild || (hasLoadingText && loading)
          const finalContainer = css`
            ${applyTagContainer(fullWidth)};
            ${applyCursor(loading ?? false, disabled ?? false)}
            ${hasPropContent
              ? applyPaddingStyle(size, variant)
              : applyWithoutTextSize(size, fullWidth)};
            ${applyShape(
              shape,
              attached ?? false,
              first ?? false,
              last ?? false,
            )};
            ${applyElementColor(variant, colorScheme, textColor)};
            ${applyBg(variant, colorScheme, backgroundColor, borderColor)};
            ${buttonRadius ? `border-radius: ${buttonRadius};` : ""}
          `

          return (
            <button
              ref={ref}
              css={css(finalContainer, sizeCss, _css)}
              disabled={disabled || loading}
              {...otherProps}
            >
              {(loading || leftIcon) && (
                <span
                  css={
                    hasPropContent
                      ? applyLeftIconStyle(size)
                      : applyIconWithoutText(size)
                  }
                >
                  {loading ? <LoadingIcon spin={true} /> : leftIcon}
                </span>
              )}
              {hasPropContent && (
                <span css={applyFontStyle(size)}>
                  {loading && loadingText ? loadingText : props.children}
                </span>
              )}
              {rightIcon && (
                <span
                  css={
                    hasPropContent
                      ? applyRightIconStyle(size)
                      : applyIconWithoutText(size)
                  }
                >
                  {rightIcon}
                </span>
              )}
            </button>
          )
        }}
      </ButtonGroupContext.Consumer>
    )
  },
)

Button.displayName = "Button"
