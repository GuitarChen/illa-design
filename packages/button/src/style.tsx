import { css, SerializedStyles } from "@emotion/react"
import {
  ButtonColorScheme,
  ButtonShape,
  ButtonSize,
  ButtonVariant,
} from "./interface"
import { globalColor, illaPrefix } from "@illa-design/theme"

// button
export function applyTagContainer(fullWidth?: boolean) {
  return css`
    ${fullWidth
      ? css`
          width: 100%;
          justify-content: center;
        `
      : null};
    transition: all 200ms ease-in-out;
    vertical-align: middle;
    white-space: nowrap;
    outline: none;
    border: 0;
    background-color: transparent;
    display: inline-flex;
    flex-direction: row;
    align-items: center;
  `
}

export enum State {
  DEFAULT,
  HOVER,
  ACTIVE,
  DISABLE,
}

export function getDifferentStatusColor(
  colorScheme: ButtonColorScheme,
  variant: ButtonVariant,
  state: State,
): string[] {
  switch (state) {
    case State.DEFAULT:
      switch (variant) {
        case "fill":
          if (colorScheme != "gray" && colorScheme != "grayBlue") {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-01`),
              globalColor(`--${illaPrefix}-white-01`),
            ]
          } else {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-09`),
              globalColor(`--${illaPrefix}-${colorScheme}-02`),
            ]
          }
        case "light": {
          return [
            globalColor(`--${illaPrefix}-${colorScheme}-07`),
            globalColor(`--${illaPrefix}-${colorScheme}-01`),
          ]
        }
        case "dashed":
        case "outline":
          if (colorScheme != "gray" && colorScheme != "grayBlue") {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-01`),
              globalColor(`--${illaPrefix}-${colorScheme}-01`),
            ]
          } else {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-08`),
              globalColor(`--${illaPrefix}-${colorScheme}-02`),
            ]
          }
        case "text":
          return [
            globalColor("transparent"),
            globalColor(`--${illaPrefix}-${colorScheme}-01`),
          ]
      }
    case State.HOVER:
      switch (variant) {
        case "fill":
          if (colorScheme != "gray" && colorScheme != "grayBlue") {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-02`),
              globalColor(`--${illaPrefix}-white-01`),
            ]
          } else {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-08`),
              globalColor(`--${illaPrefix}-${colorScheme}-02`),
            ]
          }
        case "light": {
          return [
            globalColor(`--${illaPrefix}-${colorScheme}-06`),
            globalColor(`--${illaPrefix}-${colorScheme}-01`),
          ]
        }
        case "dashed":
        case "outline":
          if (colorScheme != "gray" && colorScheme != "grayBlue") {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-02`),
              globalColor(`--${illaPrefix}-${colorScheme}-02`),
            ]
          } else {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-07`),
              globalColor(`--${illaPrefix}-${colorScheme}-02`),
            ]
          }
        case "text":
          return [
            globalColor(`--${illaPrefix}-grayBlue-09`),
            globalColor(`--${illaPrefix}-${colorScheme}-02`),
          ]
      }
    case State.ACTIVE:
      switch (variant) {
        case "fill":
          if (colorScheme != "gray" && colorScheme != "grayBlue") {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-n-01`),
              globalColor(`--${illaPrefix}-white-01`),
            ]
          } else {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-06`),
              globalColor(`--${illaPrefix}-${colorScheme}-02`),
            ]
          }
        case "light": {
          return [
            globalColor(`--${illaPrefix}-${colorScheme}-05`),
            globalColor(`--${illaPrefix}-${colorScheme}-01`),
          ]
        }
        case "dashed":
        case "outline":
          if (colorScheme != "gray" && colorScheme != "grayBlue") {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-n-01`),
              globalColor(`--${illaPrefix}-${colorScheme}-n-01`),
            ]
          } else {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-05`),
              globalColor(`--${illaPrefix}-${colorScheme}-02`),
            ]
          }
        case "text":
          return [
            globalColor(`--${illaPrefix}-grayBlue-08`),
            globalColor(`--${illaPrefix}-${colorScheme}-n-01`),
          ]
      }
    case State.DISABLE:
      switch (variant) {
        case "fill":
          if (colorScheme != "gray" && colorScheme != "grayBlue") {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-05`),
              globalColor(`--${illaPrefix}-white-01`),
            ]
          } else {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-09`),
              globalColor(`--${illaPrefix}-${colorScheme}-05`),
            ]
          }
        case "light": {
          return [
            globalColor(`--${illaPrefix}-${colorScheme}-07`),
            globalColor(`--${illaPrefix}-${colorScheme}-05`),
          ]
        }
        case "dashed":
        case "outline":
          if (colorScheme != "gray" && colorScheme != "grayBlue") {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-05`),
              globalColor(`--${illaPrefix}-${colorScheme}-05`),
            ]
          } else {
            return [
              globalColor(`--${illaPrefix}-${colorScheme}-08`),
              globalColor(`--${illaPrefix}-${colorScheme}-05`),
            ]
          }
        case "text":
          return [
            globalColor("transparent"),
            globalColor(`--${illaPrefix}-${colorScheme}-05`),
          ]
      }
  }
  return []
}

export function applyBg(
  variant: ButtonVariant,
  colorScheme: ButtonColorScheme,
  backgroundColor?: string,
  borderColor?: string,
): SerializedStyles {
  const borderCss = borderColor
    ? css`
        border: solid 1px ${borderColor};
        &:hover {
          border: solid 1px ${borderColor};
        }
        &:active {
          border: solid 1px ${borderColor};
        }
      `
    : css``
  const backgroundCss = backgroundColor
    ? css`
        background-color: ${backgroundColor};
        &:hover {
          background-color: ${backgroundColor};
        }
        &:active {
          background-color: ${backgroundColor};
        }
      `
    : css``
  switch (variant) {
    case "text":
      return css`
        &:hover {
          background-color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.HOVER,
          )[0]};
        }

        &:active {
          transition: none;
          background-color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.ACTIVE,
          )[0]};
        }
      `
    case "dashed":
      return css`
        border: dashed 1px
          ${getDifferentStatusColor(colorScheme, variant, State.DEFAULT)[0]};

        &:hover {
          border: dashed 1px
            ${getDifferentStatusColor(colorScheme, variant, State.HOVER)[0]};
        }

        &:active {
          transition: none;
          border: dashed 1px
            ${getDifferentStatusColor(colorScheme, variant, State.ACTIVE)[0]};
        }

        &:disabled {
          border: dashed 1px
            ${getDifferentStatusColor(colorScheme, variant, State.DISABLE)[0]};
        }
      `
    case "fill":
      return css`
        background-color: ${getDifferentStatusColor(
          colorScheme,
          variant,
          State.DEFAULT,
        )[0]};

        &:hover {
          background-color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.HOVER,
          )[0]};
        }

        &:active {
          transition: none;
          background-color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.ACTIVE,
          )[0]};
        }

        &:disabled {
          background-color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.DISABLE,
          )[0]};
        }
        ${borderCss}
        ${backgroundCss}
      `
    case "outline":
      return css`
        border: solid 1px
          ${getDifferentStatusColor(colorScheme, variant, State.DEFAULT)[0]};

        &:hover {
          border: solid 1px
            ${getDifferentStatusColor(colorScheme, variant, State.HOVER)[0]};
        }

        &:active {
          transition: none;
          border: solid 1px
            ${getDifferentStatusColor(colorScheme, variant, State.ACTIVE)[0]};
        }

        &:disabled {
          border: solid 1px
            ${getDifferentStatusColor(colorScheme, variant, State.DISABLE)[0]};
        }
        ${borderCss}
      `
    case "light":
      return css`
        background-color: ${getDifferentStatusColor(
          colorScheme,
          variant,
          State.DEFAULT,
        )[0]};
        &:hover {
          background-color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.HOVER,
          )[0]};
        }

        &:active {
          transition: none;
          background-color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.ACTIVE,
          )[0]};
        }

        &:disabled {
          background-color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.DISABLE,
          )[0]};
        }
      `
  }
  return css``
}

export function applyCursor(
  loading: boolean,
  disabled: boolean,
): SerializedStyles {
  if (loading) {
    return css`
      cursor: default;
    `
  } else if (disabled) {
    return css`
      cursor: not-allowed;
    `
  } else {
    return css`
      cursor: pointer;
    `
  }
}

export function applyElementColor(
  variant: ButtonVariant,
  colorScheme: ButtonColorScheme,
  textColor?: string,
): SerializedStyles {
  if (textColor) {
    return css`
      color: ${textColor};
    `
  }
  switch (variant) {
    case "text":
      return css`
        color: ${getDifferentStatusColor(
          colorScheme,
          variant,
          State.DEFAULT,
        )[1]};

        &:disabled {
          color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.DISABLE,
          )[1]};
        }
      `
    case "light":
      return css`
        color: ${getDifferentStatusColor(
          colorScheme,
          variant,
          State.DEFAULT,
        )[1]};
        &:hover {
          color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.HOVER,
          )[1]};
        }

        &:active {
          transition: none;
          color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.ACTIVE,
          )[1]};
        }
        &:disabled {
          color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.DISABLE,
          )[1]};
        }
      `
    case "outline":
    case "dashed":
      return css`
        color: ${getDifferentStatusColor(
          colorScheme,
          variant,
          State.DEFAULT,
        )[1]};

        &:hover {
          color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.HOVER,
          )[1]};
        }

        &:active {
          transition: none;
          color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.ACTIVE,
          )[1]};
        }

        &:disabled {
          color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.DISABLE,
          )[1]};
        }
      `
    case "fill":
      return css`
        color: ${getDifferentStatusColor(
          colorScheme,
          variant,
          State.DEFAULT,
        )[1]};

        &:hover {
          color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.HOVER,
          )[1]};
        }

        &:active {
          transition: none;
          color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.ACTIVE,
          )[1]};
        }

        &:disabled {
          color: ${getDifferentStatusColor(
            colorScheme,
            variant,
            State.DISABLE,
          )[1]};
        }
      `
  }
  return css``
}

export function applyShape(
  shape: ButtonShape,
  attached: boolean,
  first: boolean,
  last: boolean,
): SerializedStyles {
  switch (shape) {
    case "square":
      if (attached) {
        if (first) {
          return css`
            border-top-left-radius: 4px;
            border-bottom-left-radius: 4px;
          `
        } else if (last) {
          return css`
            border-top-right-radius: 4px;
            border-bottom-right-radius: 4px;
          `
        } else {
          return css`
            border-radius: 0;
          `
        }
      }
      return css`
        border-radius: 4px;
      `
    case "round":
      if (attached) {
        if (first) {
          return css`
            border-top-left-radius: 999px;
            border-bottom-left-radius: 999px;
          `
        } else if (last) {
          return css`
            border-top-right-radius: 999px;
            border-bottom-right-radius: 999px;
          `
        } else {
          return css`
            border-radius: 0;
          `
        }
      } else {
        return css`
          border-radius: 999px;
        `
      }
  }
}

export function applyPaddingStyle(
  size: ButtonSize,
  variant: ButtonVariant,
): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        padding: ${variant == "outline" || variant == "dashed"
          ? "1px 11px"
          : "2px 12px"};
      `
    case "medium":
      return css`
        padding: ${variant == "outline" || variant == "dashed"
          ? "4px 15px"
          : "5px 16px"};
      `
    case "large":
      return css`
        padding: ${variant == "outline" || variant == "dashed"
          ? "8px 15px"
          : "9px 16px"};
      `
  }
}

export function applyWithoutTextSize(
  size: ButtonSize,
  fullWidth?: boolean,
): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        justify-content: center;
        width: ${fullWidth ? "100%" : "24px"};
        height: 24px;
        font-size: 12px;
      `
    case "medium":
      return css`
        justify-content: center;
        width: ${fullWidth ? "100%" : "32px"};
        height: 32px;
        font-size: 12px;
      `
    case "large":
      return css`
        justify-content: center;
        width: ${fullWidth ? "100%" : "40px"};
        height: 40px;
        font-size: 12px;
      `
  }
}

export function applyFontStyle(size: ButtonSize): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        font-size: 12px;
        font-weight: normal;
        font-stretch: normal;
        font-style: normal;
        line-height: 20px;
        letter-spacing: normal;
      `
    case "medium":
    case "large":
      return css`
        font-size: 14px;
        font-weight: normal;
        font-stretch: normal;
        font-style: normal;
        line-height: 22px;
        letter-spacing: normal;
      `
  }
}

export function applyLeftIconStyle(size: ButtonSize): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        font-size: 12px;
        margin-right: 6px;
      `
    case "medium":
    case "large":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        font-size: 12px;
        margin-right: 8px;
      `
  }
}

export function applyIconWithoutText(size: ButtonSize): SerializedStyles {
  switch (size) {
    case "small":
    case "medium":
    case "large":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        justify-content: center;
      `
  }
}

export function applyRightIconStyle(size: ButtonSize): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        font-size: 12px;
        margin-left: 6px;
      `
    case "medium":
    case "large":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        font-size: 12px;
        margin-left: 8px;
      `
  }
}

// button group
export const avatarGroupContainer = css`
  display: inline-flex;
  flex-direction: row;
  align-items: center;
`

export function applySpacing(
  spacing: string,
  index: number,
  attached: boolean,
): SerializedStyles {
  if (attached) {
    return css`
      display: inline-flex;
      margin-left: ${index != 0 ? "-1px" : "0px"};

      &:hover {
        z-index: 2;
      }
    `
  } else {
    return css`
      display: inline-flex;
      margin-left: ${index == 0 ? "0px" : spacing};

      &:hover {
        z-index: 2;
      }
    `
  }
}
