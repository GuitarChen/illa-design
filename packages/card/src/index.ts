export * from "./card"
export * from "./interface"
export * from "./card-grid"
export * from "./meta"
