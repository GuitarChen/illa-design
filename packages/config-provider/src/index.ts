export * from "./config-provider"
export * from "./interface"
export * from "./config-provider-context"
export * from "./config"

// locale
export * from "./locale/en-US"
export * from "./locale/zh-CN"
export * from "./locale/def"
