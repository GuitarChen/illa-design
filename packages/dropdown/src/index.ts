export * from "./dropdown"
export * from "./interface"
