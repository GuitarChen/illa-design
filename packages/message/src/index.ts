export * from "./style"
export * from "./message"
export * from "./interface"
