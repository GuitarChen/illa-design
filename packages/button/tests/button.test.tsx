import { Button } from "../src"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import { BsApp, BsTwitch } from "react-icons/bs"

test("Button renders with text", () => {
  render(<Button data-testid="test">Hello</Button>)
  expect(screen.getByTestId("test")).toBeInTheDocument()
})

test("Button renders with different colorscheme", () => {
  render(
    <Button data-testid="test-with-gray" colorScheme="gray">
      Hello
    </Button>,
  )
  expect(screen.getByTestId("test-with-gray")).toBeInTheDocument()
  render(
    <Button data-testid="test-with-green" colorScheme="green">
      Hello
    </Button>,
  )
  expect(screen.getByTestId("test-with-green")).toBeInTheDocument()
})

test("Button renders without text", () => {
  render(<Button data-testid="test-without-text" />)
  expect(screen.getByTestId("test-without-text")).toBeInTheDocument()
  render(
    <Button
      data-testid="test-without-text-medium"
      size="medium"
      rightIcon={<BsApp />}
    />,
  )
  expect(screen.getByTestId("test-without-text-medium")).toBeInTheDocument()
  render(
    <Button
      data-testid="test-without-text-large"
      size="large"
      rightIcon={<BsApp />}
    />,
  )
  expect(screen.getByTestId("test-without-text-large")).toBeInTheDocument()
})

test("Button renders with different size", () => {
  render(
    <Button
      data-testid="test-small"
      size="small"
      rightIcon={<BsTwitch />}
      leftIcon={<BsApp />}
    >
      Hello
    </Button>,
  )
  expect(screen.getByTestId("test-small")).toBeInTheDocument()
  render(
    <Button
      data-testid="test-medium"
      size="medium"
      rightIcon={<BsTwitch />}
      leftIcon={<BsApp />}
    >
      Hello
    </Button>,
  )
  expect(screen.getByTestId("test-medium")).toBeInTheDocument()
  render(
    <Button
      data-testid="test-large"
      size="large"
      rightIcon={<BsTwitch />}
      leftIcon={<BsApp />}
    >
      Hello
    </Button>,
  )
  expect(screen.getByTestId("test-large")).toBeInTheDocument()
})

test("Button renders with different variant", () => {
  render(<Button variant="fill">Hello Fill</Button>)
  expect(screen.getByText("Hello Fill")).toMatchSnapshot()
  render(<Button variant="dashed">Hello Dashed</Button>)
  expect(screen.getByText("Hello Dashed")).toMatchSnapshot()
  render(
    <Button variant="dashed" colorScheme="gray">
      Hello Gray Dashed
    </Button>,
  )
  expect(screen.getByText("Hello Gray Dashed")).toMatchSnapshot()
  render(<Button variant="outline">Hello Outline</Button>)
  expect(screen.getByText("Hello Outline")).toMatchSnapshot()
  render(
    <Button variant="outline" colorScheme="gray">
      Hello Gray Outline
    </Button>,
  )
  expect(screen.getByText("Hello Gray Outline")).toMatchSnapshot()
  render(<Button variant="text">Hello Text</Button>)
  expect(screen.getByText("Hello Text")).toMatchSnapshot()
  render(<Button variant="light">Hello Light</Button>)
  expect(screen.getByText("Hello Light")).toMatchSnapshot()
})

test("Button renders round", () => {
  render(
    <Button data-testid="test-with-round" shape="round">
      Hello
    </Button>,
  )
  expect(screen.getByTestId("test-with-round")).toMatchSnapshot()
})

test("Button renders fullwidth", () => {
  render(
    <div
      style={{
        width: "200px",
      }}
    >
      <Button data-testid="test-with-fullwidth" fullWidth={true}>
        Hello
      </Button>
    </div>,
  )
  expect(screen.getByTestId("test-with-fullwidth")).toMatchSnapshot()
})

test("Button renders disabled", () => {
  render(
    <Button data-testid="test-button" disabled>
      Hello
    </Button>,
  )
  expect(screen.getByTestId("test-button")).toBeDisabled()
})

test("Button renders with colors", () => {
  render(
    <Button borderColor="#ff4d4d" backgroundColor="#ffcccc" textColor="#ffffff">
      Hello
    </Button>,
  )
  expect(screen.getByText("Hello").parentElement).toHaveStyle({
    color: "#ffffff",
    "border-color": "#ff4d4d",
    "background-color": "#ffcccc",
  })
})
