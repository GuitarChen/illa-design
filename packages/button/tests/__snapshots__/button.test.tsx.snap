// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders fullwidth 1`] = `
.emotion-0 {
  width: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: color 200ms ease-in-out,background-color 200ms ease-in-out;
  transition: color 200ms ease-in-out,background-color 200ms ease-in-out;
  vertical-align: middle;
  white-space: nowrap;
  outline: none;
  border: 0;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 5px 16px;
  border-radius: 8px;
  color: #fff;
  background-color: #134ae0;
}

.emotion-0:hover {
  color: #fff;
}

.emotion-0:active {
  -webkit-transition: none;
  transition: none;
  color: #fff;
}

.emotion-0:disabled {
  color: #fff;
}

.emotion-0:hover {
  background-color: #175ceb;
}

.emotion-0:active {
  -webkit-transition: none;
  transition: none;
  background-color: #083bc7;
}

.emotion-0:disabled {
  background-color: #99beff;
}

.emotion-1 {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 22px;
  letter-spacing: normal;
}

<button
  class="emotion-0"
  data-testid="test-with-fullwidth"
>
  <span
    class="emotion-1"
  >
    Hello
  </span>
</button>
`;

exports[`Button renders round 1`] = `
.emotion-0 {
  -webkit-transition: color 200ms ease-in-out,background-color 200ms ease-in-out;
  transition: color 200ms ease-in-out,background-color 200ms ease-in-out;
  vertical-align: middle;
  white-space: nowrap;
  outline: none;
  border: 0;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 5px 16px;
  border-radius: 999px;
  color: #fff;
  background-color: #134ae0;
}

.emotion-0:hover {
  color: #fff;
}

.emotion-0:active {
  -webkit-transition: none;
  transition: none;
  color: #fff;
}

.emotion-0:disabled {
  color: #fff;
}

.emotion-0:hover {
  background-color: #175ceb;
}

.emotion-0:active {
  -webkit-transition: none;
  transition: none;
  background-color: #083bc7;
}

.emotion-0:disabled {
  background-color: #99beff;
}

.emotion-1 {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 22px;
  letter-spacing: normal;
}

<button
  class="emotion-0"
  data-testid="test-with-round"
>
  <span
    class="emotion-1"
  >
    Hello
  </span>
</button>
`;

exports[`Button renders with different variant 1`] = `
.emotion-0 {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 22px;
  letter-spacing: normal;
}

<span
  class="emotion-0"
>
  Hello Fill
</span>
`;

exports[`Button renders with different variant 2`] = `
.emotion-0 {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 22px;
  letter-spacing: normal;
}

<span
  class="emotion-0"
>
  Hello Dashed
</span>
`;

exports[`Button renders with different variant 3`] = `
.emotion-0 {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 22px;
  letter-spacing: normal;
}

<span
  class="emotion-0"
>
  Hello Gray Dashed
</span>
`;

exports[`Button renders with different variant 4`] = `
.emotion-0 {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 22px;
  letter-spacing: normal;
}

<span
  class="emotion-0"
>
  Hello Outline
</span>
`;

exports[`Button renders with different variant 5`] = `
.emotion-0 {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 22px;
  letter-spacing: normal;
}

<span
  class="emotion-0"
>
  Hello Gray Outline
</span>
`;

exports[`Button renders with different variant 6`] = `
.emotion-0 {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 22px;
  letter-spacing: normal;
}

<span
  class="emotion-0"
>
  Hello Text
</span>
`;

exports[`Button renders with different variant 7`] = `
.emotion-0 {
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 22px;
  letter-spacing: normal;
}

<span
  class="emotion-0"
>
  Hello Light
</span>
`;
